### PR TITLE
[Data] add data type u16

### PR DESCRIPTION
**Pull request by ChobobDev**
# Background
from #623 

ref
[add data type u8](https://github.com/gluesql/gluesql/pull/828)

# Description
- When calculating arithmetic operations on u16 and i16, both are converted to i64 and calculated.
- For the remaining types, cast to the larger type.

# Rules
1. cast `right-hand` element into `left-hand` element type in case of arithmetics between INTEGER types 
2. in case of INTEGER * FLOAT convert to FLOAT
3. in case of INTEGER * DECIMAL convert to DECIMAL
4. in case of FLOAT * DECIMAL  convert to FLOAT

### CODE REVIEW

File: core/src/data/value/binary_op/u16.rs [hunk 1]
```diff
@@ -0,0 +1,1186 @@
...
+                .checked_add(rhs as u16)
+                .ok_or_else(|| {
+                    ValueError::BinaryOperationOverflow {
+                        lhs: U16(lhs),
+                        rhs: I64(rhs),
+                        operator: NumericBinaryOperator::Add,
+                    }
+                    .into()
+                })
+                .map(U16),
```
Comment (devgony) [inline core/src/data/value/binary_op/u16.rs:99]: Looks like we have to compute with large integer like `i64` and convert to u16 at the end like this.

```rs
I64(rhs) => {
    (lhs as i64)
        .checked_add(rhs)
        .ok_or_else(|| {
            ValueError::BinaryOperationOverflow {
                lhs: U16(lhs),
                rhs: I64(rhs),
                operator: NumericBinaryOperator::Add,
            }
            .into()
        })
        .and_then(|v| {
            v.try_into().map_err(|_| {
                ValueError::BinaryOperationOverflow {
                    lhs: U16(lhs),
                    rhs: I64(rhs),
                    operator: NumericBinaryOperator::Add,
                }
                .into()
            })
        })
        .map(U16)
```

*or we may have to fix UnaryOp logic to work after evaluation

Comment (ChobobDev) [reply core/src/data/value/binary_op/u16.rs:99]: I think the latter case is better than the former. Since it will make the code much tidier.

Comment (devgony) [reply core/src/data/value/binary_op/u16.rs:99]: We decided to throw error as soon as rhs fail while casting.
```rs
lhs.checked_add(rhs.try_into().map_err(|_| {
    ValueError::BinaryOperationOverflow {
        lhs: U16(lhs),
        rhs: I64(rhs),
        operator: NumericBinaryOperator::Add,
    }
})?)
.ok_or_else(|| {
    ValueError::BinaryOperationOverflow {
        lhs: U16(lhs),
        rhs: I64(rhs),
        operator: NumericBinaryOperator::Add,
    }
    .into()
})
.map(U16)
```

For now, It is Okay to repeat error code.
Then, refactor with macro in next PR

**Review by panarch — APPROVED** [review]
Awesome 👍 👍 
All looks nice!
Thanks for the contribution! New numeric type has come :D

**Pull request merged**